src/vm/gdbjit.cpp

```diff
@@ -1291,13 +1290,46 @@ bool NotifyGdb::BuildDebugAbbrev(MemBuf& buf)
     return true;
 }
 
+int NotifyGdb::GetArgsAndLocalsLen(NewArrayHolder<ArgsDebugInfo>& argsDebug,
+                                   unsigned int argsDebugSize,
+                                   NewArrayHolder<LocalsDebugInfo>& localsDebug,
+                                   unsigned int localsDebugSize)
+{
+    int loc_size = 0;
+    char tmp_buf[16];
+    for (int i = 0; i < argsDebugSize; i++)
+    {
+        loc_size += Leb128Encode(static_cast<int32_t>(argsDebug[i].m_native_offset), tmp_buf, sizeof(tmp_buf));
+        loc_size += 2;
+    }
+    for (int i = 0; i < localsDebugSize; i++)
+    {
+        loc_size += Leb128Encode(static_cast<int32_t>(localsDebug[i].m_native_offset), tmp_buf, sizeof(tmp_buf));
+        loc_size += 2;
+    }
+    return loc_size;
+}
+
+int NotifyGdb::GetFrameLocation(int native_offset, char*& bufVarLoc)
+{
+    char cnv_buf[16] = {0};
+    int len = Leb128Encode(static_cast<int32_t>(native_offset), cnv_buf, sizeof(cnv_buf));
+    bufVarLoc[0] = len + 1;
+    bufVarLoc[1] = DW_OP_fbreg;
+    for (int j = 0; j < len; j++)
+    {
+        bufVarLoc[j + 2] = cnv_buf[j];
+    }
+    return len + 2;
+}
 /* Build tge DWARF .debug_info section */
 bool NotifyGdb::BuildDebugInfo(MemBuf& buf, NewArrayHolder<ArgsDebugInfo> &argsDebug, unsigned int argsDebugSize,
                                NewArrayHolder<LocalsDebugInfo> &localsDebug, unsigned int localsDebugSize)
 {
 
+    int loc_size = GetArgsAndLocalsLen(argsDebug, argsDebugSize, localsDebug, localsDebugSize);
```

Comment:
It would be nice to use the same naming convention in the code - some names have underscore separated names and some use camel casing.